### PR TITLE
feat: programmatically add cms groups & perms

### PR DESCRIPTION
**Pull request by wesleyboar**
## Overview

Add management commands to create CMS groups and permissions.

<sup>This will replace the "Group Permissions" > "Instructions" from [Django CMS - Developer Guide - User Permissions / Groups / Roles](https://confluence.tacc.utexas.edu/pages/viewpage.action?pageId=250460558) which just fails so bad. To copy-paste permissions via CMS HTML is inaccurate, because the IDs of the permissions are not the same per CMS.</sup>

## Related

- permissions for [TUP-258](https://jira.tacc.utexas.edu/browse/TUP-258)

## Changes

- added [cms management command](https://docs.djangoproject.com/en/2.2/howto/custom-management-commands/) dir and doc
- added way to [programmatically set cms group permissions](https://stackoverflow.com/q/22250352/11817077)

## Testing

Follow [`taccsite_cms/management/commands/group_perms/README.md` at "Set Permissions"](https://github.com/TACC/Core-CMS/blob/task/add-mgmt-cmds-to-set-cms-group-and-perms/taccsite_cms/management/commands/group_perms/README.md#set-permissions).

## UI

console
cms admin ui

## Notes

Tested on https://dev.tup.tacc.utexas.edu via TACC/tup-ui#133.

### CODE REVIEW

File: taccsite_cms/management/commands/group_perms/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Programmatically Add Groups & Permissions to CMS
+
+## Get Permissions
+
+You may **either** download an appropriate `.html` from [Django CMS - Developer Guide - User Permissions / Groups / Roles](https://confluence.tacc.utexas.edu/x/jrntDg) **or**:
+
+1. Using the CMS admin interface, build out the permissions for a group.
+2. Using the browser Developer Tools, copy the `<option>`s from the `<select>` that ahs the permissions you chose.
```
Comment (taoteg) [inline taccsite_cms/management/commands/group_perms/README.md:8]: typo `ahs` --> `has`

File: taccsite_cms/management/commands/group_perms/README.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+## Debug Command
+
+1. Open a shell into the CMS container e.g. `docker exec -it core_cms /bin/bash`.
+2. In the shell, open a Python shell i.e. `python`.
+3. In the Python shell, run these commands:
+   1. `import os`
+   2. `import django`
+   3. `os.environ.setdefault("DJANGO_SETTINGS_MODULE", "taccsite_cms.settings")`
+   4. `django.setup()`
+   5. any lines to debug from any scripts
```
Comment (taoteg) [inline taccsite_cms/management/commands/group_perms/README.md:61]: What does this mean exactly?
Recommendation: "Any additional debugging code or scripts you want to execute."

**Review by taoteg — APPROVED** [review]
Caught a typo and suggested some clarifying wording, but looks good to me otherwise.

File: taccsite_cms/management/commands/group_perms/README.md (hunk 2 above)
Comment (wesleyboar) [inline taccsite_cms/management/commands/group_perms/README.md:61]:
```suggestion
   5. Any additional debugging code or scripts you want to execute.
```

File: taccsite_cms/management/commands/group_perms/README.md (hunk 1 above)
Comment (wesleyboar) [inline taccsite_cms/management/commands/group_perms/README.md:8]:
```suggestion
2. Using the browser Developer Tools, copy the `<option>`s from the `<select>` that has the permissions you chose.
```

**Review by wesleyboar — COMMENTED** [review]
I turned informal recommendations into committable suggestion comments.

File: taccsite_cms/management/commands/group_perms/README.md (hunk 2 above)
Comment (wesleyboar) [reply taccsite_cms/management/commands/group_perms/README.md]: It means what you thought. I have updated the sentence to exactly your sentence (3664a07).

Do I even need this step?

File: taccsite_cms/management/commands/group_perms/README.md (hunk 1 above)
Comment (wesleyboar) [reply taccsite_cms/management/commands/group_perms/README.md]: Thanks. Fixed (cfbf680).

**Review by rstijerina — APPROVED** [review]

**Pull request merged**